x/sequencer/keeper/unbond.go

```diff
@@ -66,52 +88,9 @@ func (k Keeper) forceUnbondSequencer(ctx sdk.Context, seqAddr string) error {
 		k.removeUnbondingSequencer(ctx, seq)
 	}
 
-	ctx.EventManager().EmitEvent(
-		sdk.NewEvent(
-			types.EventTypeUnbonded,
-			sdk.NewAttribute(types.AttributeKeySequencer, seqAddr),
-			sdk.NewAttribute(types.AttributeKeyBond, seqTokens.String()),
-		),
-	)
-
-	return nil
-}
-
-// unbondUnbondingSequencer unbonds a sequencer that currently unbonding
-func (k Keeper) unbondUnbondingSequencer(ctx sdk.Context, seqAddr string) error {
-	seq, found := k.GetSequencer(ctx, seqAddr)
-	if !found {
-		return types.ErrUnknownSequencer
-	}
-
-	if seq.Status != types.Unbonding {
-		return errorsmod.Wrapf(
-			types.ErrInvalidSequencerStatus,
-			"sequencer status is not unbonding: got %s",
-			seq.Status.String(),
-		)
+	if seq.IsNoticePeriodInProgress() {
+		k.removeNoticePeriodSequencer(ctx, seq)
```

Comment:
I don't get why this is here because they should never be able to unbond gracefully if notice period is not finished

if this is just for slash unbond, then it should be in caller

Reply:
it's the actual unbond method. it should leave the store clean and insync